docs/publishing.md

```diff
@@ -0,0 +1,105 @@
+# How to publish Jenkins agent deb PPA
+
+This document will talk about how to bootstrap a debian package repo, what are its components, 
+what they do and how to build, test, sign and publish a debian package. 
```

Comment:
I think this sentence would work better as a bulleted list. Something like:

> This document describes various aspects of bootstrapping a debian package repository, including:
> * The repository's components 
> * The purpose of each component 
> * How to build, test, sign and publish a debian package

Feel free to edit as you see fit.